clients/wallet/src/resubmissions.rs

```diff
@@ -0,0 +1,795 @@
+							tracing::debug!(
+								"auto_check_failed_tasks(): Removing from cache: {xdr}"
+							);
+							wallet.remove_tx_envelope_from_cache(seq);
+						},
+					}
+				}
+			}
+		}
+	});
+}
+
+#[cfg(test)]
+mod test {
+	use crate::{
+		error::Error, mock::*, operations::create_basic_spacewalk_stellar_transaction,
+		resubmissions::pause_process_in_secs, StellarWallet,
+	};
+	use mocktopus::mocking::{MockResult, Mockable};
+	use primitives::{
+		stellar::{types::AlphaNum4, Asset as StellarAsset, TransactionEnvelope, XdrCodec},
+		TransactionEnvelopeExt,
+	};
+	use serial_test::serial;
+
+	#[tokio::test]
+	#[serial]
+	async fn check_is_transaction_already_submitted() {
+		let wallet = wallet_with_storage("resources/check_is_transaction_already_submitted")
+			.expect("")
+			.clone();
+		let mut wallet = wallet.write().await;
+
+		let asset = StellarAsset::native();
+		let amount = 1002;
+
+		// test is_transaction_already_submitted returns true
+		{
+			let response = wallet
+				.send_payment_to_address(
+					default_destination(),
+					asset.clone(),
+					amount,
+					rand::random(),
+					DEFAULT_STROOP_FEE_PER_OPERATION,
+					false,
+				)
+				.await
+				.expect("should be ok");
+
+			let tx_envelope = TransactionEnvelope::from_base64_xdr(response.envelope_xdr)
+				.expect("should return an envelope");
+			let tx = tx_envelope.get_transaction().expect("should return a transaction");
+
+			// check that the transaction truly exists
+			assert!(wallet.is_transaction_already_submitted(&tx).await);
+		}
+
+		// test is_transaction_already_submitted returns false
+		{
+			let dummy_tx = create_basic_spacewalk_stellar_transaction(
+				rand::random(),
+				DEFAULT_STROOP_FEE_PER_OPERATION,
+				wallet.public_key(),
+				1,
+			)
+			.expect("return a transaction");
+
+			assert!(!wallet.is_transaction_already_submitted(&dummy_tx).await);
+		}
+
+		wallet.remove_cache_dir();
+	}
+
+	#[tokio::test]
+	#[serial]
+	async fn check_bump_sequence_number_and_submit() {
+		let wallet = wallet_with_storage("resources/check_bump_sequence_number_and_submit")
+			.expect("should return a wallet")
+			.clone();
+		let wallet = wallet.write().await;
+		let seq = wallet.get_sequence().await.expect("return sequence number");
+
+		let asset = StellarAsset::native();
+		let amount = 1002;
+
+		// test bump_sequence_number_and_submit success
+		{
+			let dummy_envelope = wallet
+				.create_payment_envelope(
+					default_destination(),
+					asset.clone(),
+					amount,
+					rand::random(),
+					DEFAULT_STROOP_FEE_PER_OPERATION,
+					seq - 5,
+				)
+				.expect("should return an envelope");
+
+			let mut dummy_transaction =
+				dummy_envelope.get_transaction().expect("must return a transaction");
+
+			let _ = wallet
+				.bump_sequence_number_and_submit(dummy_transaction.clone())
+				.await
+				.expect("return ok");
+
+			dummy_transaction.seq_num = seq + 1;
+
+			assert!(wallet.is_transaction_already_submitted(&dummy_transaction).await);
```

Comment:
Nit: Since `is_transaction_already_submitted` does not check the sequence number of the transaction we wouldn't even need to increment it beforehand.

Reply:
removed